[1.14.0] Tests fail on big-endian platforms

When compiling version 1.14.0 for (64-bit) big-endian platforms (s390x, ppc64), many tests error or fail in **cdrizzle.tridz()**, f.e.:
```
_______________________ test_zero_input_weight[lanczos3] _______________________

kernel = 'lanczos3'

    @pytest.mark.parametrize(
        'kernel', ['square', 'point', 'turbo', 'gaussian', 'lanczos3'],
    )
    def test_zero_input_weight(kernel):
        """
        Test do_driz square kernel with grid
        """
        # initialize input:
        insci = np.ones((200, 400), dtype=np.float32)
        inwht = np.ones((200, 400), dtype=np.float32)
        inwht[:, 150:155] = 0
    
        # initialize output:
        outsci = np.zeros((210, 410), dtype=np.float32)
        outwht = np.zeros((210, 410), dtype=np.float32)
        outctx = np.zeros((210, 410), dtype=np.int32)
    
        # define coordinate mapping:
        pixmap = np.moveaxis(np.mgrid[1:201, 1:401][::-1], 0, -1)
    
        # resample:
>       cdrizzle.tdriz(
            insci, inwht, pixmap,
            outsci, outwht, outctx,
            uniqid=1,
            xmin=0, xmax=400,
            ymin=0, ymax=200,
            pixfrac=1,
            kernel=kernel,
            in_units='cps',
            expscale=1,
            wtscale=1,
            fillstr='INDEF',
        )
E       ValueError: No or too few valid pixels in the pixel map.

drizzle/tests/test_drizzle.py:276: ValueError
```
This is a regression; it didn't happen with version 1.13.7. 
Versions:
* Python 3.11.4
* Astropy 5.3.3
* Numpy 1.24.2

[Full build log on s390x](https://buildd.debian.org/status/fetch.php?pkg=python-drizzle&arch=s390x&ver=1.14.0-1&stamp=1694783686&raw=0)

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -4,9 +4,13 @@
 Release Notes
 =============
 
-.. 1.14.1 (unreleased)
+.. 1.14.2 (unreleased)
    ===================
 
+1.14.1 (2023-09-16)
+===================
+
+- Addressed test failures on big endian architectures. [#116]
 
 
 1.14.0 (2023-09-15)

--- a/src/cdrizzleapi.c
+++ b/src/cdrizzleapi.c
@@ -59,11 +59,11 @@ tdriz(PyObject *obj UNUSED_PARAM, PyObject *args, PyObject *keywords)
 
   /* Arguments in the order they appear */
   PyObject *oimg, *owei, *pixmap, *oout, *owht, *ocon;
-  long uniqid = 1;
-  long xmin = 0;
-  long xmax = 0;
-  long ymin = 0;
-  long ymax = 0;
+  integer_t uniqid = 1;
+  integer_t xmin = 0;
+  integer_t xmax = 0;
+  integer_t ymin = 0;
+  integer_t ymax = 0;
   double scale = 1.0;
   double pfract = 1.0;
   char *kernel_str = "square";
@@ -89,9 +89,9 @@ tdriz(PyObject *obj UNUSED_PARAM, PyObject *args, PyObject *keywords)
   driz_log_message("starting tdriz");
   driz_error_init(&error);
 
-  if (!PyArg_ParseTupleAndKeywords(args, keywords, "OOOOOO|lllllddssffs:tdriz", (char **)kwlist,
+  if (!PyArg_ParseTupleAndKeywords(args, keywords, "OOOOOO|iiiiiddssffs:tdriz", (char **)kwlist,
                         &oimg, &owei, &pixmap, &oout, &owht, &ocon, /* OOOOOO */
-                        &uniqid, &xmin, &xmax, &ymin, &ymax,  /* lllll */
+                        &uniqid, &xmin, &xmax, &ymin, &ymax,  /* iiiii */
                         &scale, &pfract, &kernel_str, &inun_str, /* ddss */
                         &expin, &wtscl,  &fillstr) /* ffs */
                        ) {

